lang/test/org/partiql/lang/eval/EvaluatingCompilerDateTimeTests.kt

```diff
@@ -236,5 +236,52 @@ class EvaluatingCompilerDateTimeTests : EvaluatorTestBase() {
             assertEquals(expected, originalExprNode.toString())
         }
     }
+    @ParameterizedTest
+    @ArgumentsSource(ArgumentsForComparison::class)
+    fun testComparison(tc: ComparisonTestCase) {
+        when (tc.throws) {
+            true ->
+                try {
+                    voidEval(tc.query)
+                    fail("Expected ${tc.query} to throw an error")
+                } catch(e: EvaluationException) {
+                    // EvaluationException is thrown as expected, do nothing.
+                }
+            false -> {
+                val originalExprValue = eval(tc.query)
+                assertEquals(tc.expected, originalExprValue.toString())
+            }
+        }
+    }
+
+    data class ComparisonTestCase(val query: String, val expected: String?, val throws: Boolean)
```

Comment:
I would not make the `throws` a boolean, you should have it be an exception class or something more explicit.  Also, it is the case that one cannot have an `expected` non-null `String` **and** a `throws`, yes?  I would document that.

Reply:
Makes sense. Generally we check for the `ErrorCode` when the `Exception` is thrown. However this error doesn't have one. I have removed the `throws` boolean and documented it well to indicate that the `null` `expected` string value throws an error.